Modified: trunk/src/ide/external/cython/Cython/Distutils/build_ext.py
==============================================================================
--- trunk/src/ide/external/cython/Cython/Distutils/build_ext.py	(original)
+++ trunk/src/ide/external/cython/Cython/Distutils/build_ext.py	Thu Jul 22 16:17:50 2010
@@ -195,5 +195,10 @@
                     generate_pxi = pyrex_gen_pxi)
                 result = cython_compile(source, options=options,
                                         full_module_name=module_name)
+                if os.path.isfile(target):
+                    os.chmod(target, 0444)
+                if result.num_errors != 0:
+                    raise DistutilsExecError("cython c/c++ generation failed")
+                    
 
         return new_sources
